cmake/GenerateGrpcSources.cmake

```diff
@@ -0,0 +1,21 @@
+set(proto_srcs_dir "${PROJECT_BINARY_DIR}/proto")
+
+include(ImportGrpc)
+
+#----------------------------------------------------------------------
+# Generate sources from proto files
+#----------------------------------------------------------------------
+function(GenerateGrpcSources proto_file proto_path proto_srcs proto_hdrs grpc_srcs grpc_hdrs)
+  get_filename_component(proto_out_path "${proto_srcs}" PATH)
+  add_custom_command(
+    OUTPUT "${proto_srcs}" "${proto_hdrs}" "${grpc_srcs}" "${grpc_hdrs}"
+    COMMAND ${_PROTOBUF_PROTOC}
+    ARGS --grpc_out "${proto_out_path}"
+      --cpp_out "${proto_out_path}"
+      -I "${proto_path}"
+      -I ${CMAKE_SOURCE_DIR}/third_party/grpc/third_party/protobuf/src/
+      -I ${CMAKE_SOURCE_DIR}/source/protobuf
```

Comment:
Similarly to above, make sure we're using the right value between `${CMAKE_SOURCE_DIR}` and `${CMAKE_CURRENT_SOURCE_DIR}` based on your intended behavior.

Reply:
To be clear, I have trouble fully understanding when to use each so this may be correct as it is.

Reply:
Yeah, these will be `CMAKE_SOURCE_DIR` to get back to the root even when they're included from elsewhere.